### DISCUSSION TITLE
Two Uart Implementations 'Exiting with abort signal and core dump file. \ Aborted'

### DISCUSSION THREAD

**Discussion by sw3llz** [nasa/fprime#2888] [General]
Howdy - 

Im currently setting up 2 uart connections with fprime. 

one is the xbee implementation and another is for an eps impl. 

I am able to run the Drv::LinuxUartDriver as expected  (with only the xbee connection) on fprime 3.4.3. I'm creating two Fw::Buffer objects for both the xbee and the eps impl, and am able to open the ports for /dev/ttyUSB0 and /dev/ttyACM0. I tried the fix mentioned in this thread ( https://github.com/nasa/fprime/discussions/2071#discussioncomment-6126324 ) - and previously, i was getting an error where there wasnt any bins to assign (this was before allocating a buffer object for the eps uart connection. 

I spent a some hours fighting with the autocoder in the generated build directory, adding 'state' to configComponents as mentioned in this thread ( https://github.com/nasa/fprime/discussions/2814#discussioncomment-10188978 ) - and editing the deploymentTopologyAc(.cpp/.hpp). 

Attached are photos that help convey the state of my issue. 

I've edited the deploymentTopology.cpp and have tried phases - both together and separately.

Is there a way to connect 2 uart lines with 2 Fw::Buffer objects ? Could I be going about this any better? 

![Screenshot from 2024-09-21 12-58-40](https://github.com/user-attachments/assets/aabf0804-1325-4266-b0fe-67a4c733b0aa)
![Screenshot from 2024-09-21 12-59-15](https://github.com/user-attachments/assets/51ca2425-0d32-4647-bf09-1fd0f351e54e)
![Screenshot from 2024-09-21 13-00-10](https://github.com/user-attachments/assets/b6a960f2-7b32-4bdd-b76b-f94ae53d0b1c)
![Screenshot from 2024-09-21 13-00-37](https://github.com/user-attachments/assets/a25251b7-63fd-4a84-8d7b-01c996dfd40c)
![Screenshot from 2024-09-21 13-01-22](https://github.com/user-attachments/assets/2de282c3-fec7-4ed0-b5d5-eedea1973755)

**Comment by thomas-bc** [comment]
https://github.com/nasa/fprime/blob/dfc655f7892802e5ecfcd9e9a9c133c76ef2aa59/config/FpConfig.h#L232

Making this config value larger in your config will allow the assert string to be longer, and we'll be able to see where the assertion is happening in the file.

**Reply by LeStarch** [reply]
You may also need to increase the values below too (if you are using the assert fatal adapter).   Change all the string sizes to match your change above.

https://github.com/nasa/fprime/blob/v3.4.3/Svc/AssertFatalAdapter/AssertFatalEvents.fppi

This will be fixed in the next release.

**Reply by sw3llz** [reply]
My apologies, that seemed to add additional information. Im a bit confused about how the drivers are able to configure and open the ports but they are returning a null pointer with the assert calls. Any thoughts? 

![image](https://github.com/user-attachments/assets/d7d21299-91b7-49d2-860e-68ec82563224)

**Comment by sw3llz** [comment]
I used this as a reference https://github.com/fprime-community/fprime-osr/blob/ctrl-algorithm/OsrDeployment/Top/topology.fpp 

.. it's working now
